src/mlpack/tests/main_tests/linear_svm_test.cpp

```diff
@@ -0,0 +1,893 @@
+/**
+  * @file logistic_regression_test.cpp
+  * @author Yashwant Singh Parihar
+  *
+  * Test mlpackMain() of logistic_regression_main.cpp
+  *
+  * mlpack is free software; you may redistribute it and/or modify it under the
+  * terms of the 3-clause BSD license.  You should have received a copy of the
+  * 3-clause BSD license along with mlpack.  If not, see
+  * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+  */
+#include <string>
+
+#define BINDING_TYPE BINDING_TYPE_TEST
+
+static const std::string testName = "LinearSVM";
+
+#include <mlpack/core.hpp>
+#include <mlpack/methods/linear_svm/linear_svm_main.cpp>
+#include <mlpack/core/util/mlpack_main.hpp>
+#include "test_helper.hpp"
+
+#include <boost/test/unit_test.hpp>
+#include "../test_tools.hpp"
+
+using namespace mlpack;
+
+
+struct LinearSVMTestFixture
+{
+ public:
+  LinearSVMTestFixture()
+  {
+    // Cache in the options for this program.
+    CLI::RestoreSettings(testName);
+  }
+
+  ~LinearSVMTestFixture()
+  {
+    // Clear the settings.
+    bindings::tests::CleanMemory();
+    CLI::ClearSettings();
+  }
+};
+
+BOOST_FIXTURE_TEST_SUITE(LinearSVMMainTest,
+                         LinearSVMTestFixture);
+
+/**
+ * Ensure that trainingSet are necessarily passed when training.
+ **/
+BOOST_AUTO_TEST_CASE(LinearSVMNoTrainingData)
+{
+  arma::Row<size_t> trainLabels;
+  if (!data::Load("iris_labels.txt", trainLabels))
+    BOOST_FAIL("Cannot load test dataset iris_labels.txt!");
+
+  SetInputParam("labels", std::move(trainLabels));
+
+  // Training data is not provided. Should throw a runtime error.
+  Log::Fatal.ignoreInput = true;
+  BOOST_REQUIRE_THROW(mlpackMain(), std::runtime_error);
+  Log::Fatal.ignoreInput = false;
+}
+
+/**
+ * Checking that that size and dimensionality of prediction is correct.
+ */
+BOOST_AUTO_TEST_CASE(LinearSVMOutputDimensionTest)
+{
+  arma::mat trainData;
+  if (!data::Load("iris.csv", trainData))
+    BOOST_FAIL("Cannot load test dataset iris.csv!");
+
+  arma::Row<size_t> trainLabels;
+  if (!data::Load("iris_labels.txt", trainLabels))
+    BOOST_FAIL("Cannot load test dataset iris_labels.txt!");
+
+  arma::mat testData;
+  if (!data::Load("iris_test.csv", testData))
+    BOOST_FAIL("Cannot load test dataset iris_test.csv!");
+
+  size_t testSize = testData.n_cols;
+
+  SetInputParam("training", std::move(trainData));
+  SetInputParam("labels", std::move(trainLabels));
+  SetInputParam("test", std::move(testData));
+
+  // Training the model.
+  mlpackMain();
+
+  // Get the output predictions of the test data.
+  const arma::Row<size_t> &testLabels =
```

Comment:
This should be `arma::Row<size_t>& testLabels`, not `arma::Row<size_t> &testLabels`.